src/hooks/useDataMutation.tsx

```diff
@@ -7,6 +7,11 @@ import { useState } from "react";
 import { useFetch } from "@hooks/useFetch";
 import { FetchResponse } from "@models/response/response";
 
+type RequestVariables<T> = {
+  body?: T;
+  headersInit?: HeadersInit;
```

Comment:
Could be just be just me, but I can only read this in a british accent 🤣. 
Might be worth renaming to `InitialHeaders`